### PR TITLE
Issue 14028 - [CTFE] Possible reinterpret cast to a pointer to static array

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14028

Use AddrExp + SliceExp (&aggr[a..b]) to represent a CTFE pointer to static array.

### CODE REVIEW

**Comment by don-clugston-sociomantic** [comment]
LGTM. And thanks for your continued work on the CTFE engine in my absence. In particular getting rid of NeedLValueRef was a great step forward.

File: src/interpret.c [hunk 1]
```diff
@@ -2083,15 +2083,33 @@ class Interpreter : public Visitor
...
                 result = new AddrExp(e->loc, val);
                 result->type = e->type;
                 return;
             }
+
+            // It's OK to cast from fixed length to fixed length array, eg &int[n] to int[d]*.
+            if (val->type->ty == Tsarray && pointee->ty == Tsarray &&
+                elemsize == pointee->nextOf()->size())
+            {
+                size_t d = ((TypeSArray *)pointee)->dim->toInteger();
```
Comment (yebblies) [inline src/interpret.c]: Cast please.

Comment (9rnsr) [inline src/interpret.c]: Fixed.

**Comment by 9rnsr** [comment]
@don-clugston-sociomantic My pleasure. Now I'm working to fix some CTFE issues.

Ready to merge.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**